### PR TITLE
RFC: Change codegen names

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: devito/types/dense.py [hunk 1]
```diff
@@ -588,11 +588,11 @@ def symbolic_shape(self):
     _C_typename = 'struct %s *' % _C_structname
     _C_field_data = 'data'
     _C_field_size = 'size'
-    _C_field_nopad_size = 'npsize'
-    _C_field_domain_size = 'dsize'
-    _C_field_halo_size = 'hsize'
-    _C_field_halo_ofs = 'hofs'
-    _C_field_owned_ofs = 'oofs'
+    _C_field_nopad_size = 'size_nopad'
```
Comment (georgebisbas) [inline devito/types/dense.py]: Its up to you but I strongly disagree with 'size_xxx'
It is also weird to have 
 _C_field_xxx_size = 'size_xxx'
Doesn't make sense IMHO

File: devito/types/dense.py [hunk 2]
```diff
@@ -588,11 +588,11 @@ def symbolic_shape(self):
...
+    _C_field_domain_size = 'size_domain'
+    _C_field_halo_size = 'size_halo'
+    _C_field_owned_ofs = 'ofs_owned'
```
Comment (georgebisbas) [inline devito/types/dense.py]: The same...

**Comment by FabioLuporini** [comment]
the more time we wait, the more the conflicts we will have to fix

**Comment by FabioLuporini** [comment]
people feel free to reopen if interested in pursuing this

**Pull request closed without merging**